nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -116,6 +117,27 @@ in {
       '';
     };
 
+    phpExtraExtensions = mkOption {
+      type = with types; mkOptionType {
+        name = "selectorFunction";
+        description = "Function that selects PHP extensions from the list of available ones";
+        check = isFunction;
+        merge = _location: selectors: all: concatLists (map (selector: selector.value all) selectors);
+        emptyValue = _all: [];
+      };
```

Comment:
I'm wondering if that should be factored out into `lib/types.nix` (cc @Infinisil )

Reply:
Would be possible. Is there other usages of a similar concept somewhere?

Reply:
Oh yes definitely, a function type is something we still need. See https://github.com/NixOS/nixpkgs/pull/44601

Reply:
Great to see that this pops up in other places as well. But that PR seems a bit stuck. Is it worth waiting for it, or should we proceed with what's here, and refactor later?

Reply:
Hmm, how about changing this to just `listOf str`? E.g. `phpExtraExtensions = [ "bz2" ]`.

Reply:
I'd find it confusing. Especially if the usual way to select PHP extensions is with a selector function. I'd rather make good documentation to explain this pattern well, because it's ubiquitous in nixpkgs.

Reply:
Yeah sounds good. Then I think we can roll with this custom type here for now and once above PR is merged (or one like it, more likely) replace it